drud/ddev
# Add ddev share command, fixes #375

## The Problem/Issue/Bug:

#375 requests ngrok for sharing a view of a project. 

## How this PR Solves The Problem:

Spawn ngrok with proper config to share config, using `ddev share`

## Manual Testing Instructions:

* Install ngrok
* Create an account on ngrok.com
* Set up, https://dashboard.ngrok.com/get-started
* Start a project and `ddev share`
* Experiment with additional args other than --subdomain by adding them to `ngrok_args` in .ddev/config.yaml
* Consider a paid account to experiment with additional features (required for --subdomain)
* Experiment with `ddev share` with no account (Just comment out the token in ~/.ngrok2/ngrok.yml). It should work, but with a warning.

## Automated Testing Overview:

* TestShareCmd tests basic unauthenticated use of `ddev share`

## Related Issue Link(s):

OP #375 

## Release/Deployment notes:

The testbots have to have ngrok on them, they were updated by this PR. First time buildkite testbot maintenance via PR.
